### PR TITLE
Add error listener

### PR DESCRIPTION
**Pull request by sonh**
Need to handle the error occurs when lost connection.

### CODE REVIEW

File: src/main/java/org/stellar/sdk/requests/EventListener.java [hunk 1]
```diff
@@ -9,4 +9,6 @@
    * @param object object deserialized from the event data
    */
   void onEvent(T object);
+
+  void onError();
```
Comment (jillesvangurp) [inline src/main/java/org/stellar/sdk/requests/EventListener.java]: pass the exception as a parameter; probably a common use case for onError will be logging

File: src/main/java/org/stellar/sdk/requests/SSEStream.java [hunk 2]
```diff
@@ -172,11 +172,14 @@ public void onFailure(EventSource eventSource, @Nullable Throwable t, @Nullable
       if (t != null) {
         if (t instanceof SocketException) {
           // not a failure, server disconnected
+          listener.onError();
```
Comment (jillesvangurp) [inline src/main/java/org/stellar/sdk/requests/SSEStream.java]: we probably should have custom callbacks for expected events like this

**Comment by jillesvangurp** [comment]
StreamingSmokeTest is broken. Otherwise liking the idea of callbacks instead of throwing exceptions that are swallowed. I do think we should have a couple of different callbacks for expected situations like the server disconnecting after a while.

**Comment by sonhvp** [comment]
#175
Currently, we will get a crash if lost connection. Instead of throwing exceptions, we should have some callbacks to handle exceptions.

**Pull request closed without merging**